### ISSUE TITLE
Populating the Title attribute with the header text

### ISSUE THREAD

**Issue by gjones** [tscanlin/tocbot#233]
Hi, 
Great work on Tocbot, I found it really easy to implement and to customise. Many thanks for your work.

One thing that has come up is that often my header titles can get pretty pretty long. In the toc I'm truncating them so they don't span multiple lines. It would be really great if the links automatically included the link title as the html title attribute on the link. This allows visitors to see the full title when they hover over the truncated one.  

So instead of: 
```<a href="#my-long-link" class="toc-button node-name--H2">My really, really, really long headline</a>```

You'd have:
```<a href="#my-long-link" class="toc-button node-name--H2" title="My really, really, really long headline">My really, really, really long headline</a>```

I'm guessing it might be as simple as adding the following to build-html.js, but didn't want to presume. 

```
a.setAttribute('href', options.basePath + '#' + data.id)
a.setAttribute('class', options.linkClass +
   SPACE_CHAR + 'node-name--' + data.nodeName +
   SPACE_CHAR + options.extraLinkClasses)
****
a.setAttribute('title', data.textContent)
**** 
item.appendChild(a)
 return item
}
```

Thanks again!

**Comment by tscanlin** [comment]
Hey, thanks for opening this issue and sorry for the late reply, I just got back from a vacation and connectivity was spotty at times.

I think this is a great idea though! Would you be willing to make a PR for this change? If not I will try to get to it later this week.

**Comment by gjones** [comment]
@tscanlin I've just added a PR now, there's probably some cleaning up required, but hopefully it saves you some time. Thanks!

**Comment by tscanlin** [comment]
This is published in `tocbot@4.13.0`. Thank you for the contribution @gjones!
